sdcm/provision/azure/provisioner.py

```diff
@@ -58,12 +58,18 @@ def __post_init__(self) -> None:
             self._cache[v_m.name] = self._vm_to_instance(v_m)
 
     @classmethod
-    def discover_regions(cls, test_id: str) -> List["AzureProvisioner"]:
+    def discover_regions(cls, test_id: str = "") -> List["AzureProvisioner"]:
+        """Discovers provisioners for in each region for given test id.
+
+        If test_id is not provided, it discovers all related to SCT provisioners."""
         all_resource_groups = AzureService().resource.resource_groups.list()
         test_resource_groups = [rg for rg in all_resource_groups if rg.name.startswith(f"SCT-{test_id}")]
         provisioners = []
         for resource_group in test_resource_groups:
-            provisioners.append(cls(test_id, resource_group.location))
+            tid = test_id if test_id else "-".join(resource_group.name.split("-")[1:-1])
+            if not tid:
+                continue
+            provisioners.append(cls(tid, resource_group.location))
```

Comment:
Did you thought about handling non-empty `test_id` first:

```python
if test_id:
    return [cls(test_id, rg.location) for rg in all_resource_groups if rg.name.startswith(f"SCT-{test_id}")]
```

?

Reply:
I don't remember :)
Thanks for pointing it out. Solution wasn't that clear and I think I've found cleaner/easier one now.